### DISCUSSION TITLE
WebGL for NanoVDB and examples

### DISCUSSION THREAD

**Discussion by roman01la** [AcademySoftwareFoundation/openvdb#1486] [General]
Hello! Not sure if this is the best place to ask, but I was wondering if WebGL support has already landed and if yes, are there any examples?

**Comment by d-simon** [comment]
Bump – I'm also interested in this.

**Comment by amesgames** [comment]
I am also interested and am embarking on this effort myself.

I will post back to this thread if any progress is made.

Perhaps we can collaborate.

**Comment by d-simon** [comment]
I did find this implementation: https://github.com/mjurczyk/openvdb

**Comment by S0ulfire84** [comment]
I'm wondering about this too. The only thing I've seen is this thread here, which seems to imply it should almost be finished: https://groups.google.com/g/openvdb-forum/c/-JY6pfcvU7M

**Comment by trsh** [comment]
As for the GPU side. WebGL doesn't support compute shaders. I really doubt you can pull this of with VS and PS pipeline. CUDA is not for web's. 
As for the CPU side, I can confirm it's possible with Emscripten, but will require minor changes to code and very-right settings for the compiler.

**Comment by emcfarlane** [comment]
I put together a POC for NanoVDB in WebGPU: https://github.com/emcfarlane/webgpu-nanovdb

It's a port of the PNanoVDB.h into WGSL. Theres some limitations in WGSL that make it difficult to decode the volume (lack of 64-bit ints is the main one). Hopefully someone finds it helpful!
